src/Compilers/Core/Portable/DiagnosticAnalyzer/AnalyzerExecutor.cs

```diff
@@ -1238,7 +1233,7 @@ private Action<Diagnostic> GetAddDiagnostic(ISymbol contextSymbol, ImmutableArra
         {
             return diagnostic =>
             {
-                if (isGeneratedCodeSymbol && shouldSuppressGeneratedCodeDiagnostic(diagnostic, analyzer, compilation, cancellationToken))
+                if (shouldSuppressGeneratedCodeDiagnostic(diagnostic, analyzer, compilation, cancellationToken))
```

Comment:
> tree can be cached whether it is generated code or not (lazily). 

We already do that - http://source.roslyn.io/Microsoft.CodeAnalysis/R/3256449c00c25c6f.html. However, that map is only based on file extensions. We can consider caching spans for generated code within trees which don't have generated code extensions, but may have symbols with GeneratedCodeAttribute - basically lazily build a map from tree to symbols with generated code attribute. Let me do that change and update the PR